temporal-testing/src/main/java/io/temporal/testing/TestActivityEnvironmentInternal.java

```diff
@@ -95,6 +95,7 @@ public final class TestActivityEnvironmentInternal implements TestActivityEnviro
   private final TestEnvironmentOptions testEnvironmentOptions;
   private final WorkflowServiceStubs workflowServiceStubs;
   private ClassConsumerPair<Object> activityHeartbeatListener;
+  private Object heartbeatDetails;
```

Comment:
Could you replace it with AtomicReference and read-following-with-write in executeActivity with `getAndSet`?
It's mostly to eliminate a possibility of a write happened in `setHeartbeatDetails` being overridden with null without ever being read if there is a race.

Reply:
Makes sense, thanks! New commit should address